### PR TITLE
[FEATURE] Create designtime and support bundles for libraries

### PR DESCRIPTION
**Pull request by codeworrior**
When a bundle would be empty (not contain any modules), it will be skipped.

### CODE REVIEW

File: lib/tasks/bundlers/generateLibraryPreload.js [hunk 1]
```diff
@@ -2,10 +2,24 @@ const log = require("@ui5/logger").getLogger("builder:tasks:bundlers:generateLib
...
+	".js",
+	".control.xml", // XMLComposite
+	".fragment.html",
+	".fragment.json",
+	".fragment.xml",
+	".view.html",
+	".view.json",
+	".view.xml",
+	".properties",
+	".json"
```
Comment (codeworrior) [inline lib/tasks/bundlers/generateLibraryPreload.js:15]: The general ".json" file type is NOT present in our Maven tooling. I've kept it nevertheless for compatibility with existing projects that use the ui5-tooling already.

File: lib/tasks/bundlers/generateLibraryPreload.js [hunk 2]
```diff
@@ -247,22 +343,11 @@ module.exports = function({workspace, dependencies, options: {projectName}}) {
...
-						}).then(([bundle]) => {
-							if (bundle) {
-								// console.log(`${libraryNamespace}/library-preload.js bundle created`);
-								return workspace.write(bundle);
-							}
-						});
+						return createLibraryBundles(libraryNamespace, resources)
+							.then((results) => {
+								const bundles = Array.prototype.concat.apply([], results);
+								return Promise.all(bundles.map((bundle) => bundle && workspace.write(bundle)));
```
Comment (codeworrior) [inline lib/tasks/bundlers/generateLibraryPreload.js:349]: Due to the new `skipIfEmpty`, some of the entries in the `bundles` array might be `undefined`.

**Comment by RandomByte** [comment]
I like this

File: lib/tasks/bundlers/generateLibraryPreload.js [hunk 3]
```diff
@@ -2,10 +2,24 @@ const log = require("@ui5/logger").getLogger("builder:tasks:bundlers:generateLib
...
+	".view.xml",
+	".properties",
+	".json"
+];
+
 function getDefaultLibraryPreloadFilters(namespace) {
 	return [
 		`${namespace}/`,
 		`!${namespace}/.library`,
+		`!${namespace}/*-preload.js`, // exclude all bundles
```
Comment (matz3) [inline lib/tasks/bundlers/generateLibraryPreload.js:22]: This does not exclude the `designtime/library-preload.designtime.js` or `library-preload.support.js`. Is this intended?

Comment (codeworrior) [reply lib/tasks/bundlers/generateLibraryPreload.js:22]: But the excludes in line 23 and 24 both will exclude it, don't they?

Comment (matz3) [reply lib/tasks/bundlers/generateLibraryPreload.js:22]: Oh yeah, I missed that. It's friday 😄

**Review by RandomByte — APPROVED** [review]

**Pull request merged**